pkg/reconciler/openshift/tektonaddon/tektonaddon.go

```diff
@@ -416,41 +410,27 @@ func createInstallerSet(ctx context.Context, oc clientset.Interface, ta *v1alpha
 		return err
 	}
 
-	is := makeInstallerSet(ta, manifest, installerSetPrefix, releaseVersion, specHash)
-
-	createdIs, err := oc.OperatorV1alpha1().TektonInstallerSets().
-		Create(ctx, is, metav1.CreateOptions{})
-	if err != nil {
-		return err
-	}
-
-	if len(ta.Status.AddonsInstallerSet) == 0 {
-		ta.Status.AddonsInstallerSet = map[string]string{}
-	}
-
-	// Update the status of addon with created installerSet name
-	ta.Status.AddonsInstallerSet[component] = createdIs.Name
-	ta.Status.SetVersion(releaseVersion)
+	is := makeInstallerSet(ta, manifest, installerSetPrefix, releaseVersion, component, specHash)
 
-	_, err = oc.OperatorV1alpha1().TektonAddons().
-		UpdateStatus(ctx, ta, metav1.UpdateOptions{})
-	if err != nil {
+	if _, err := oc.OperatorV1alpha1().TektonInstallerSets().
+		Create(ctx, is, metav1.CreateOptions{}); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func makeInstallerSet(ta *v1alpha1.TektonAddon, manifest mf.Manifest, prefix, releaseVersion, specHash string) *v1alpha1.TektonInstallerSet {
+func makeInstallerSet(ta *v1alpha1.TektonAddon, manifest mf.Manifest, prefix, releaseVersion, component, specHash string) *v1alpha1.TektonInstallerSet {
```

Comment:
here also let us pass the set of labes as a single argument (label selector itself or map). 
so that formatiing the labels will be the responsibility of the caller.